providers/inwx/inwxProvider.go

```diff
@@ -0,0 +1,292 @@
+
+var features = providers.DocumentationNotes{
+	providers.CanUseAlias:            providers.Cannot("INWX does not support the ALIAS or ANAME record type."),
+	providers.CanUseCAA:              providers.Can(),
+	providers.CanUseDS:               providers.Unimplemented("DS records require a different API call that hasn't been implemented yet."),
+	providers.CanUsePTR:              providers.Can(),
+	providers.CanUseNAPTR:            providers.Can(),
+	providers.CanUseSRV:              providers.Can("SRV records with empty targets are not supported."),
+	providers.CanUseSSHFP:            providers.Can(),
+	providers.CanUseTLSA:             providers.Can(),
+	providers.CanUseTXTMulti:         providers.Cannot("INWX only supports a single entry for TXT records"),
+	providers.CanAutoDNSSEC:          providers.Unimplemented("Supported by INWX but not implemented yet."),
+	providers.DocOfficiallySupported: providers.Cannot(),
+	providers.DocDualHost:            providers.Can(),
+	providers.DocCreateDomains:       providers.Unimplemented("Supported by INWX but not implemented yet."),
+	providers.CanGetZones:            providers.Can(),
+	providers.CanUseAzureAlias:       providers.Cannot(),
+}
+
+func init() {
+	providers.RegisterRegistrarType("INWX", newInwxReg)
+	providers.RegisterDomainServiceProviderType("INWX", newInwxDsp, features)
+}
+
+func newInwx(m map[string]string) (*InwxApi, error) {
+	for key := range m {
+		switch key {
+		case "username",
+			"password",
+			"totp",
+			"sandbox",
+			"domain":
+			continue
+		default:
+			return nil, fmt.Errorf("INWX: unknown key in `creds.json` (%s)\n", key)
+		}
+	}
+
+	if m["username"] == "" {
+		return nil, fmt.Errorf("INWX Username must be provided.")
+	}
+	if m["password"] == "" {
+		return nil, fmt.Errorf("INWX Password must be provided.")
+	}
+
+	var sandbox bool
+	if m["sandbox"] == "1" {
+		sandbox = true
+	} else {
+		sandbox = false
+	}
+
+	opts := &goinwx.ClientOptions{Sandbox: sandbox}
+	client := goinwx.NewClient(m["username"], m["password"], opts)
+
+	err := client.Account.Login()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to login to INWX")
+	}
+
+	if m["totp"] != "" {
+		err := client.Account.Unlock(m["totp"])
+		if err != nil {
+			return nil, fmt.Errorf("Could not unlock INWX account - TOTP is probably invalid.")
+		}
+	}
+
+	api := &InwxApi{client: client, sandbox: sandbox}
+
+	return api, nil
+}
+
+func newInwxReg(m map[string]string) (providers.Registrar, error) {
+	return newInwx(m)
+}
+
+func newInwxDsp(m map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
+	return newInwx(m)
+}
+
+func makeNameserverRecordRequest(domain string, rec *models.RecordConfig) *goinwx.NameserverRecordRequest {
+	req := &goinwx.NameserverRecordRequest{
+		Domain:  domain,
+		Type:    rec.Type,
+		Content: rec.GetTargetField(),
+		Name:    rec.GetLabel(),
+		Ttl:     int(rec.TTL),
+	}
+	targetWithoutDot := strings.TrimRight(rec.GetTargetField(), ".")
```

Comment:
Humor me and try this to see if it works:

    t := rec.GetTargetField()
    targetWithoutDot :=t[:len(t)-2]

If that doesn't work, TrimSuffix instead of TrimRight.

Why?  (1) TrimRight() will remove multiple dots.  (2) GetTargetField() should be consistent in always returning a trailing dot.

Reply:
Good point, I've replaced it with `[:len(t)-1]`